### ISSUE TITLE
Unserialize error

### ISSUE THREAD

**Issue by klimenttoshkov** [zerodahero/laravel-workflow#38]
Hi again,

I have stumbled upon this error:
```
[2021-02-06 17:02:46] local.ERROR: unserialize(): Error at offset 3683 of 3922 bytes {"exception":"[object] (ErrorException(code: 0): unserialize(): Error at offset 3683 of 3922 bytes at /Users/sag/Dropbox/code/api_v3/vendor/zerodahero/laravel-workflow/src/Events/BaseEvent.php:56)
[stacktrace]
#0 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError(8, 'unserialize(): ...', '/Users/sag/Drop...', 56, Array)
#1 /Users/sag/Dropbox/code/api_v3/vendor/zerodahero/laravel-workflow/src/Events/BaseEvent.php(56): unserialize('O:37:\"Modules\\\\B...')
#2 [internal function]: ZeroDaHero\\LaravelWorkflow\\Events\\BaseEvent->unserialize('a:5:{s:16:\"base...')
#3 /Users/sag/Dropbox/code/api_v3/vendor/cloudcreativity/laravel-json-api/src/Queue/UpdateClientProcess.php(64): unserialize('O:36:\"Illuminat...')
#4 /Users/sag/Dropbox/code/api_v3/vendor/cloudcreativity/laravel-json-api/src/Queue/UpdateClientProcess.php(37): CloudCreativity\\LaravelJsonApi\\Queue\\UpdateClientProcess->deserialize(Object(Illuminate\\Queue\\Jobs\\RedisJob))
#5 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Events/Dispatcher.php(412): CloudCreativity\\LaravelJsonApi\\Queue\\UpdateClientProcess->handle(Object(Illuminate\\Queue\\Events\\JobFailed))
#6 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Events/Dispatcher.php(237): Illuminate\\Events\\Dispatcher->Illuminate\\Events\\{closure}('Illuminate\\\\Queu...', Array)
#7 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(195): Illuminate\\Events\\Dispatcher->dispatch('Illuminate\\\\Queu...')
#8 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(544): Illuminate\\Queue\\Jobs\\Job->fail(Object(ErrorException))
#9 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(505): Illuminate\\Queue\\Worker->failJob(Object(Illuminate\\Queue\\Jobs\\RedisJob), Object(ErrorException))
#10 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(433): Illuminate\\Queue\\Worker->markJobAsFailedIfWillExceedMaxAttempts('redis', Object(Illuminate\\Queue\\Jobs\\RedisJob), 1, Object(ErrorException))
#11 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(410): Illuminate\\Queue\\Worker->handleJobException('redis', Object(Illuminate\\Queue\\Jobs\\RedisJob), Object(Illuminate\\Queue\\WorkerOptions), Object(ErrorException))
#12 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(356): Illuminate\\Queue\\Worker->process('redis', Object(Illuminate\\Queue\\Jobs\\RedisJob), Object(Illuminate\\Queue\\WorkerOptions))
#13 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(307): Illuminate\\Queue\\Worker->runJob(Object(Illuminate\\Queue\\Jobs\\RedisJob), 'redis', Object(Illuminate\\Queue\\WorkerOptions))
#14 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(116): Illuminate\\Queue\\Worker->runNextJob('redis', 'bsms', Object(Illuminate\\Queue\\WorkerOptions))
#15 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(100): Illuminate\\Queue\\Console\\WorkCommand->runWorker('redis', 'bsms')
#16 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Illuminate\\Queue\\Console\\WorkCommand->handle()
#17 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Container/Util.php(40): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#18 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure(Object(Closure))
#19 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Array, Object(Closure))
#20 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Container/Container.php(610): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Array, Array, NULL)
#21 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Console/Command.php(136): Illuminate\\Container\\Container->call(Array)
#22 /Users/sag/Dropbox/code/api_v3/vendor/symfony/console/Command/Command.php(256): Illuminate\\Console\\Command->execute(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Illuminate\\Console\\OutputStyle))
#23 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Console/Command.php(121): Symfony\\Component\\Console\\Command\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Illuminate\\Console\\OutputStyle))
#24 /Users/sag/Dropbox/code/api_v3/vendor/symfony/console/Application.php(971): Illuminate\\Console\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#25 /Users/sag/Dropbox/code/api_v3/vendor/symfony/console/Application.php(290): Symfony\\Component\\Console\\Application->doRunCommand(Object(Illuminate\\Queue\\Console\\WorkCommand), Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#26 /Users/sag/Dropbox/code/api_v3/vendor/symfony/console/Application.php(166): Symfony\\Component\\Console\\Application->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#27 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Console/Application.php(93): Symfony\\Component\\Console\\Application->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#28 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(129): Illuminate\\Console\\Application->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#29 /Users/sag/Dropbox/code/api_v3/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#30 {main}

[previous exception] [object] (ErrorException(code: 0): unserialize(): Error at offset 3683 of 3922 bytes at /Users/sag/Dropbox/code/api_v3/vendor/zerodahero/laravel-workflow/src/Events/BaseEvent.php:56)
[stacktrace]
#0 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError(8, 'unserialize(): ...', '/Users/sag/Drop...', 56, Array)
#1 /Users/sag/Dropbox/code/api_v3/vendor/zerodahero/laravel-workflow/src/Events/BaseEvent.php(56): unserialize('O:37:\"Modules\\\\B...')
#2 [internal function]: ZeroDaHero\\LaravelWorkflow\\Events\\BaseEvent->unserialize('a:5:{s:16:\"base...')
#3 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(95): unserialize('O:36:\"Illuminat...')
#4 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(251): Illuminate\\Queue\\CallQueuedHandler->getCommand(Array)
#5 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(213): Illuminate\\Queue\\CallQueuedHandler->failed(Array, Object(ErrorException), '7dada2bb-c2f3-4...')
#6 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(192): Illuminate\\Queue\\Jobs\\Job->failed(Object(ErrorException))
#7 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(544): Illuminate\\Queue\\Jobs\\Job->fail(Object(ErrorException))
#8 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(505): Illuminate\\Queue\\Worker->failJob(Object(Illuminate\\Queue\\Jobs\\RedisJob), Object(ErrorException))
#9 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(433): Illuminate\\Queue\\Worker->markJobAsFailedIfWillExceedMaxAttempts('redis', Object(Illuminate\\Queue\\Jobs\\RedisJob), 1, Object(ErrorException))
#10 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(410): Illuminate\\Queue\\Worker->handleJobException('redis', Object(Illuminate\\Queue\\Jobs\\RedisJob), Object(Illuminate\\Queue\\WorkerOptions), Object(ErrorException))
#11 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(356): Illuminate\\Queue\\Worker->process('redis', Object(Illuminate\\Queue\\Jobs\\RedisJob), Object(Illuminate\\Queue\\WorkerOptions))
#12 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(307): Illuminate\\Queue\\Worker->runJob(Object(Illuminate\\Queue\\Jobs\\RedisJob), 'redis', Object(Illuminate\\Queue\\WorkerOptions))
#13 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(116): Illuminate\\Queue\\Worker->runNextJob('redis', 'bsms', Object(Illuminate\\Queue\\WorkerOptions))
#14 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(100): Illuminate\\Queue\\Console\\WorkCommand->runWorker('redis', 'bsms')
#15 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Illuminate\\Queue\\Console\\WorkCommand->handle()
#16 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Container/Util.php(40): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#17 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure(Object(Closure))
#18 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Array, Object(Closure))
#19 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Container/Container.php(610): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Array, Array, NULL)
#20 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Console/Command.php(136): Illuminate\\Container\\Container->call(Array)
#21 /Users/sag/Dropbox/code/api_v3/vendor/symfony/console/Command/Command.php(256): Illuminate\\Console\\Command->execute(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Illuminate\\Console\\OutputStyle))
#22 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Console/Command.php(121): Symfony\\Component\\Console\\Command\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Illuminate\\Console\\OutputStyle))
#23 /Users/sag/Dropbox/code/api_v3/vendor/symfony/console/Application.php(971): Illuminate\\Console\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#24 /Users/sag/Dropbox/code/api_v3/vendor/symfony/console/Application.php(290): Symfony\\Component\\Console\\Application->doRunCommand(Object(Illuminate\\Queue\\Console\\WorkCommand), Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#25 /Users/sag/Dropbox/code/api_v3/vendor/symfony/console/Application.php(166): Symfony\\Component\\Console\\Application->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#26 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Console/Application.php(93): Symfony\\Component\\Console\\Application->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#27 /Users/sag/Dropbox/code/api_v3/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(129): Illuminate\\Console\\Application->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#28 /Users/sag/Dropbox/code/api_v3/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#29 {main}
"} 
```

I am trying to process Queued subscriber the same way as I have already did with many others. But for some reason only this subscriber is failing. There is nothing different in model, workflow definition, nor subscriber. I have recorded the serialized data in BaseEvent and then manually unserialized it without problem. 

Processing the events in synchronous manner does not reproduce the problem (removing `implements ShouldQueue` from subscriber definition) but I want to process it in queue.

Can you help? I am sure something is different and I am overlooking it, but I can't find the problem by myself.

workflow definition:
```
    'bsmsDlrOutgoing' => [
        'supports' => [
            OutgoingDlr::class,
        ],
        'places' => [
            'new',
            'processing',
        ],
        'initial_places' => [
            'new'
        ],
        'transitions' => [
            'toProcessing' => [
                'from' => 'new',
                'to' => 'processing',
            ],
        ],
    ],
```

model definition:
```
<?php

namespace Modules\BSMS\Entities\Cdr;

use Illuminate\Database\Eloquent\Model;
use Modules\BSMS\Entities\Route\Provider\Provider;
use ZeroDaHero\LaravelWorkflow\Traits\WorkflowTrait;

class OutgoingDlr extends Model
{
    use WorkflowTrait;

    protected $table = 'api_v3_bsms.cdr_out_dlr';

    protected $fillable = [
        'data',
    ];

    protected $casts = [
        'marking' => 'array',
        'data' => 'array',
    ];

    public function message()
    {
        return $this->belongsTo(Outgoing::class, 'cdr_out_id');
    }

    public function provider()
    {
        return $this->belongsTo(Provider::class, 'provider_id');
    }
}
```

subscriber:
```
<?php

namespace Modules\BSMS\Listeners;

use Illuminate\Queue\InteractsWithQueue;
use Illuminate\Contracts\Queue\ShouldQueue;
use Illuminate\Database\Eloquent\ModelNotFoundException;
use Illuminate\Events\Dispatcher;
use Illuminate\Support\Facades\App;
use Illuminate\Support\Facades\Log;
use ZeroDaHero\LaravelWorkflow\Events\EnteredEvent;

class OutgoingDlrAsyncSubscriber implements ShouldQueue
{
    use InteractsWithQueue;

    public $queue = 'bsms';
    public $tries = 1;
    public $maxExceptions = 1;
    public $timeout = 20;

    public function anyAny($name, array $payload)
    {
        Log::debug('Received event: ' . $name);

        // $event = $payload[0];
        // $subject = $event->getSubject();
    }

    public function enteredProcessing(EnteredEvent $event)
    {
        $subject = $event->getSubject();
        $subject->save();
    }

    public function subscribe(Dispatcher $events)
    {
        $events->listen('workflow.bsmsDlrOutgoing.*.*', [self::class, 'anyAny']);
        // $events->listen('workflow.bsmsDlrOutgoing.entered.processing', [self::class, 'enteredProcessing']);

    }
}
```

**Comment by klimenttoshkov** [comment]
Found the problem but don't know how to mitigate it.

Provider/Provider is `class Provider extends Authenticatable` and for some reason fails serialization.
But I need this relation to exist in order to further process the workflow data.

Edit: Actually any kind of ->associate()-> relations in the model cause Queued subscribers to fail!

**Comment by klimenttoshkov** [comment]
To reproduce the problem:

```
$creator = $request->user();
$a = new Model([...]);
$a->createdBy()->associate($creator);
$a->workflow_apply('toSomething');
```

If you have non-queued listener, everything goes fine.
But if your listener is queued (`implements ShouldQueue`) then it fails deserialisation. 
Serialised data itself appears fine to me, i was able to record it in BaseEvent to text file and then unserialise it. I can provide samples.

**Comment by zerodahero** [comment]
Hi! Thanks for all the detail on this! This is a bit strange. Are you able to serialize and unserialize the model on its own with those associations? The authenticatable class _should_ still be serializeable, but I wonder if one of the hash characters is messing with things? Is `password` in the `$hidden` property?

Also, what are you using for queue driver? If it serializes and unserializes (with the association) OK outside of the queue, there might be a character set or encoding issue?

I'll try and see if I can recreate this on my end.

**Comment by klimenttoshkov** [comment]
Have not tried to serialize them on their own, never felt urge to...
The problem is not related to authenticatable. At first I thought this was the only difference, but then found out that the problem comes with associating. I could reproduce it with regular Model. This rejects the probability that some hashed passwords may be the source of problem.

Queue driver is redis (using predis). I use it a lot, never had problems with it before.Here is dump from the data received at BaseEvent:

```
Array
(
    [base_event_class] => Symfony\Component\Workflow\Event\GuardEvent
    [subject] => O:37:"Modules\BSMS\Entities\Cdr\OutgoingDlr":27:{s:8:".*.table";s:23:"api_v3_bsms.cdr_out_dlr";s:11:".*.fillable";a:1:{i:
0;s:4:"data";}s:8:".*.casts";a:1:{s:4:"data";s:5:"array";}s:13:".*.connection";s:5:"mysql";s:13:".*.primaryKey";s:2:"id";s:10:".*.keyType"
;s:3:"int";s:12:"incrementing";b:1;s:7:".*.with";a:0:{}s:12:".*.withCount";a:0:{}s:10:".*.perPage";i:15;s:6:"exists";b:1;s:18:"wasRecently
Created";b:1;s:13:".*.attributes";a:6:{s:4:"data";s:2:"[]";s:11:"provider_id";i:2;s:10:"updated_at";s:19:"2021-02-06 16:35:46";s:10:"creat
ed_at";s:19:"2021-02-06 16:35:46";s:2:"id";i:8;s:7:"marking";a:1:{s:3:"new";i:1;}}s:11:".*.original";a:5:{s:4:"data";s:2:"[]";s:11:"provid
er_id";i:2;s:10:"updated_at";s:19:"2021-02-06 16:35:46";s:10:"created_at";s:19:"2021-02-06 16:35:46";s:2:"id";i:8;}s:10:".*.changes";a:0:{
}s:17:".*.classCastCache";a:0:{}s:8:".*.dates";a:0:{}s:13:".*.dateFormat";N;s:10:".*.appends";a:0:{}s:19:".*.dispatchesEvents";a:0:{}s:14:
".*.observables";a:0:{}s:12:".*.relations";a:1:{s:8:"provider";O:45:"Modules\BSMS\Entities\Route\Provider\Provider":29:{s:8:".*.table";s:2
1:"api_v3_bsms.providers";s:11:".*.fillable";a:0:{}s:8:".*.casts";a:1:{s:6:"active";s:7:"boolean";}s:13:".*.connection";s:5:"mysql";s:13:"
.*.primaryKey";s:2:"id";s:10:".*.keyType";s:3:"int";s:12:"incrementing";b:1;s:7:".*.with";a:0:{}s:12:".*.withCount";a:0:{}s:10:".*.perPage
";i:15;s:6:"exists";b:1;s:18:"wasRecentlyCreated";b:0;s:13:".*.attributes";a:4:{s:2:"id";i:2;s:10:"account_id";i:3;s:11:"driver_name";s:3:
"nth";s:6:"active";i:1;}s:11:".*.original";a:4:{s:2:"id";i:2;s:10:"account_id";i:3;s:11:"driver_name";s:3:"nth";s:6:"active";i:1;}s:10:".*
.changes";a:0:{}s:17:".*.classCastCache";a:0:{}s:8:".*.dates";a:0:{}s:13:".*.dateFormat";N;s:10:".*.appends";a:0:{}s:19:".*.dispatchesEven
ts";a:0:{}s:14:".*.observables";a:0:{}s:12:".*.relations";a:0:{}s:10:".*.touches";a:0:{}s:10:"timestamps";b:1;s:9:".*.hidden";a:0:{}s:10:"
.*.visible";a:0:{}s:10:".*.guarded";a:1:{i:0;s:1:"*";}s:20:".*.rememberTokenName";s:14:"remember_token";s:14:".*.accessToken";O:35:"Larave
l\Sanctum\PersonalAccessToken":27:{s:8:".*.casts";a:2:{s:9:"abilities";s:4:"json";s:12:"last_used_at";s:8:"datetime";}s:11:".*.fillable";a
:3:{i:0;s:4:"name";i:1;s:5:"token";i:2;s:9:"abilities";}s:9:".*.hidden";a:1:{i:0;s:5:"token";}s:13:".*.connection";s:5:"mysql";s:8:".*.tab
le";s:22:"personal_access_tokens";s:13:".*.primaryKey";s:2:"id";s:10:".*.keyType";s:3:"int";s:12:"incrementing";b:1;s:7:".*.with";a:0:{}s:
12:".*.withCount";a:0:{}s:10:".*.perPage";i:15;s:6:"exists";b:1;s:18:"wasRecentlyCreated";b:0;s:13:".*.attributes";a:9:{s:2:"id";i:3;s:14:
"tokenable_type";s:45:"Modules\BSMS\Entities\Route\Provider\Provider";s:12:"tokenable_id";i:2;s:4:"name";s:3:"dlr";s:5:"token";s:64:"7b18b
6ce95a5e545f8b040f67db869ecc2c00cd37c1d3e8c9332955a4f1cf837";s:9:"abilities";s:5:"["*"]";s:12:"last_used_at";s:19:"2021-02-06 16:35:46";s:
10:"created_at";s:19:"2021-02-05 13:39:37";s:10:"updated_at";s:19:"2021-02-06 16:35:46";}s:11:".*.original";a:9:{s:2:"id";i:3;s:14:"tokena
ble_type";s:45:"Modules\BSMS\Entities\Route\Provider\Provider";s:12:"tokenable_id";i:2;s:4:"name";s:3:"dlr";s:5:"token";s:64:"7b18b6ce95a5
e545f8b040f67db869ecc2c00cd37c1d3e8c9332955a4f1cf837";s:9:"abilities";s:5:"["*"]";s:12:"last_used_at";s:19:"2021-02-06 16:35:46";s:10:"cre
ated_at";s:19:"2021-02-05 13:39:37";s:10:"updated_at";s:19:"2021-02-06 16:35:46";}s:10:".*.changes";a:2:{s:12:"last_used_at";s:19:"2021-02
-06 16:35:46";s:10:"updated_at";s:19:"2021-02-06 16:35:46";}s:17:".*.classCastCache";a:0:{}s:8:".*.dates";a:0:{}s:13:".*.dateFormat";N;s:1
0:".*.appends";a:0:{}s:19:".*.dispatchesEvents";a:0:{}s:14:".*.observables";a:0:{}s:12:".*.relations";a:1:{s:9:"tokenable";r:45;}s:10:".*.
touches";a:0:{}s:10:"timestamps";b:1;s:10:".*.visible";a:0:{}s:10:".*.guarded";a:1:{i:0;s:1:"*";}}}}s:10:".*.touches";a:0:{}s:10:"timestam
ps";b:1;s:9:".*.hidden";a:0:{}s:10:".*.visible";a:0:{}s:10:".*.guarded";a:1:{i:0;s:1:"*";}}
    [marking] => O:34:"Symfony\Component\Workflow\Marking":1:{s:42:".Symfony\Component\Workflow\Marking.places";a:1:{s:3:"new";i:1;}}
    [transition] => O:37:"Symfony\Component\Workflow\Transition":3:{s:43:".Symfony\Component\Workflow\Transition.name";s:12:"toProcessing"
;s:44:".Symfony\Component\Workflow\Transition.froms";a:1:{i:0;s:3:"new";}s:42:".Symfony\Component\Workflow\Transition.tos";a:1:{i:0;s:10:"
processing";}}
    [workflow] => Array
        (
            [name] => bsmsDlrOutgoing
        )

)
```

If I paste it in unserialize function it works fine. At the same time throws error in Laravel.

**Comment by klimenttoshkov** [comment]
Did a quick test and can confirm that the Model can be serialized/deserialized on it's own without problems:

```
        $m = OutgoingDlr::find (39);        
        $c = Outgoing::find(51);
        
        $m->message()->associate($c);

        $mSerialized = serialize($m);
        $mUnSerialized = unserialize($mSerialized);

        dd ($mUnSerialized);
```

But processing it in queue fails.

**Comment by zerodahero** [comment]
Hmmm, ok, must be related to the base event, that does some manual serialization/deserialization to cope with a closure in the Symfony component.

**Comment by lixiaohui100200** [comment]
请问有线上的例子可以看吗？

**Comment by zerodahero** [comment]
Hi, I was able to get a few moments to take a closer look (sorry, it's been a busy week). So far, I'm unable to reproduce this in my test setup, but I also haven't fully built it up as you've described.

Looking at the data you sent and the error, I'm thinking it's related to the double serialization going on. I've removed that in an experimental branch, here: https://github.com/zerodahero/laravel-workflow/tree/serialize-experiment If you get a chance, try that out and see if it resolves the issue.

In the data you gave, roughly around the error offset there's a serialized reference, which I think is causing the issue. The only other thing I could think of would potentially be different encodings, but I'm not seeing anything like that in your info.
